pkg/agent/datapath/policyBridge.go

```diff
@@ -0,0 +1,459 @@
+	ctCommitAction := ofctrl.NewConntrackAction(true, false, &sfcPolicyTable, &policyConntrackZone)
+	ctCommitFlow.SetConntrack(ctCommitAction)
+
+	ctCommitTableDefaultFlow, _ := p.ctCommitTable.NewFlow(ofctrl.FlowMatch{
+		Priority: DEFAULT_FLOW_PRIORITY,
+	})
+	ctCommitTableDefaultFlow.Next(p.sfcPolicyTable)
+
+	// sfc policy table
+	sfcPolicyTableDefaultFlow, _ := p.sfcPolicyTable.NewFlow(ofctrl.FlowMatch{
+		Priority: DEFAULT_FLOW_PRIORITY - 3,
+	})
+	sfcPolicyTableDefaultFlow.Next(p.policyForwardingTable)
+
+	// policy forwarding table
+	fromLocalOutputFlow, _ := p.policyForwardingTable.NewFlow(ofctrl.FlowMatch{
+		Priority:  NORMAL_MATCH_FLOW_PRIORITY,
+		InputPort: uint32(POLICY_TO_LOCAL_PORT),
+		Regs: []*ofctrl.NXRegister{
+			{
+				RegID: 6,
+				Data:  0,
+				Range: openflow13.NewNXRange(0, 15),
+			},
+		},
+	})
+	outputPort, _ = p.ofSwitch.OutputPort(POLICY_TO_CLS_PORT)
+	fromLocalOutputFlow.Next(outputPort)
+
+	fromUpstreamOuputFlow, _ := p.policyForwardingTable.NewFlow(ofctrl.FlowMatch{
+		Priority:  NORMAL_MATCH_FLOW_PRIORITY,
+		InputPort: uint32(POLICY_TO_CLS_PORT),
+		Regs: []*ofctrl.NXRegister{
+			{
+				RegID: 6,
+				Data:  0,
+				Range: openflow13.NewNXRange(0, 15),
+			},
+		},
+	})
+	outputPort, _ = p.ofSwitch.OutputPort(POLICY_TO_LOCAL_PORT)
+	fromUpstreamOuputFlow.Next(outputPort)
+
+	return nil
+}
+
+func (p *PolicyBridge) BridgeReset() error {
+	return nil
+}
+
+func (p *PolicyBridge) AddLocalEndpoint(endpoint *Endpoint) error {
+	return nil
+}
+
+func (p *PolicyBridge) RemoveLocalEndpoint(endpoint *Endpoint) error {
+	return nil
+}
+
+func (p *PolicyBridge) GetTierTable(direction uint8, tier uint8) (*ofctrl.Table, *ofctrl.Table, error) {
+	var policyTable, nextTable *ofctrl.Table
+	switch direction {
+	case POLICY_DIRECTION_OUT:
+		switch tier {
+		case POLICY_TIER0:
+			policyTable = p.egressTier1PolicyTable
+			nextTable = p.egressTier2PolicyTable
+		case POLICY_TIER1:
+			policyTable = p.egressTier2PolicyTable
+			nextTable = p.egressTier3PolicyTable
+		case POLICY_TIER2:
+			policyTable = p.egressTier3PolicyTable
+			nextTable = p.ctCommitTable
+		default:
+			return nil, nil, errors.New("unknow policy tier")
+		}
+	case POLICY_DIRECTION_IN:
+		switch tier {
+		case POLICY_TIER0:
+			policyTable = p.ingressTier1PolicyTable
+			nextTable = p.ingressTier2PolicyTable
+		case POLICY_TIER1:
+			policyTable = p.ingressTier2PolicyTable
+			nextTable = p.ingressTier3PolicyTable
+		case POLICY_TIER2:
+			policyTable = p.ingressTier3PolicyTable
+			nextTable = p.ctCommitTable
+		default:
+			return nil, nil, errors.New("unknow policy tier")
+		}
+	}
+
+	return policyTable, nextTable, nil
+}
+
+func (p *PolicyBridge) AddMicroSegmentRule(rule *EveroutePolicyRule, direction uint8, tier uint8) (*ofctrl.Flow, error) {
+	var ipDa *net.IP = nil
+	var ipDaMask *net.IP = nil
+	var ipSa *net.IP = nil
+	var ipSaMask *net.IP = nil
+	var err error
+
+	// make sure switch is connected
+	if !p.isPolicySwitchConnected {
```

Comment:
Should grab the Mutex to access the isPolicySwitchConnected? Or use the IsSwitchConnected function.